### PR TITLE
[KAFKA-8820] [WIP] Use Admin API of Replica Reassignment in CLI tools

### PR DESCRIPTION
**Pull request by steverod**
Utilize the new Admin API for replica reassignment in the command-line tools, as described in KIP-455: https://cwiki.apache.org/confluence/display/KAFKA/KIP-455%3A+Create+an+Administrative+API+for+Replica+Reassignment

This particular PR is a first PR for Kafka and is only a change to the kafka-topics.sh code to print out  adding/removing replicas. It includes both a TopicCommandTest and a TopicCommandWithAdminClientTest. Since the new replica reassignment API isn't getting used, we  are expecting no replicas to actually be 
*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/main/scala/kafka/admin/TopicCommand.scala [hunk 1]
```diff
@@ -138,13 +140,20 @@ object TopicCommand extends Logging {
       !hasLeader || !liveBrokers.contains(info.leader.id)
     }
 
+    def hasReassigningReplicas: Boolean = {
+      addingReplicas.nonEmpty || removingReplicas.nonEmpty
+    }
+
+
```
Comment (stanislavkozlovski) [inline core/src/main/scala/kafka/admin/TopicCommand.scala:147]: nit: extra space

File: core/src/test/scala/unit/kafka/admin/TopicCommandWithAdminClientTest.scala [hunk 2]
```diff
@@ -717,6 +717,40 @@ class TopicCommandWithAdminClientTest extends KafkaServerTestHarness with Loggin
...
+        Array("--topic", reassigningTopic, "--replica-assignment", "4:1:0", "--partitions", "6")))
+
+      val aliveServers = servers.filterNot(_.config.brokerId == 0)
+      TestUtils.waitUntilMetadataIsPropagated(aliveServers, reassigningTopic, 0)
+      val output = TestUtils.grabConsoleOutput(
+        topicService.describeTopic(new TopicCommandOptions(Array())))
+      val rows = output.split("\n")
+      assertTrue(rows(0).startsWith(s"\tTopic: $reassigningTopic"))
+      assertTrue(rows(1).startsWith(s"\tTopic: $regularTopic"))
+      assertEquals(2, rows.length);
```
Comment (stanislavkozlovski) [inline core/src/test/scala/unit/kafka/admin/TopicCommandWithAdminClientTest.scala:748]: Are we validating the reassignment info here?

File: core/src/test/scala/unit/kafka/admin/TopicCommandTest.scala [hunk 3]
```diff
@@ -519,6 +519,38 @@ class TopicCommandTest extends ZooKeeperTestHarness with Logging with RackAwareT
...
+
+    // Now alter the partition assignment and get a description.
+    topicService.alterTopic(new TopicCommandOptions(
+      Array("--topic", testTopicName, "--replica-assignment", "2:4:1")))
+    // XXX: How do I avoid the race of this completing trivially?
+    val alteredOutputWithConfig = TestUtils.grabConsoleOutput(describeTopicsWithConfig())
+    assertTrue(alteredOutputWithConfig.contains(testTopicName) && !alteredOutputWithConfig.contains(reassignmentAddingDescribe))
+    assertTrue(alteredOutputWithConfig.contains(testTopicName) && !alteredOutputWithConfig.contains(reassignmentRemovingDescribe))
+
+
```
Comment (stanislavkozlovski) [inline core/src/test/scala/unit/kafka/admin/TopicCommandTest.scala:550]: nit: extra space

**Comment by gwenshap** [comment]
Should we test cancellation?

**Comment by cmccabe** [comment]
The reassign-partitions.sh command was upgraded to use the `--bootstrap-server` flag in #8244

**Pull request closed without merging**